### PR TITLE
buildsystem: update to gcc (to 7.2), glibc and binutils

### PR DESCRIPTION
**Pull request by InuSasha**
This is a WIP. Not all packages/addons are building, actual.

I believe, same commit are not related to the gcc7 change. But i had no time to confirm.
Feel free to cherry-pick ;)

packages and logs under https://dl.inusasha.de/libreelec/dev/gcc7/

**Not building addons in PR:**
none

**Not building packages in PR & Master:**
kodi (imx6)
alsa-lib (imx6)
qemu (Generic)

**Not building addons in PR & Master:**
adsp.biquad.filters
audiodecoder.asap
audiodecoder.upse
audiodecoder.usf
audiodecoder.wsr
game.libretro
game.libretro.mame2010
game.libretro.mame2014
game.libretro.mupen64plus
hyperion
peripheral.steamcontroller
peripheral.xarcade
pvr.octonet
screensaver.asteroids
screensaver.asterwave
screensaver.biogenesis
screensaver.cpblobs
screensaver.greynetic
screensaver.matrixtrails
screensaver.pingpong
screensaver.pyro
screensaver.shadertoy
screensavers.rsxs
screensaver.stars
visualization.vsxu
visualization.wavforhue

### CODE REVIEW

**Comment by chewitt** [comment]
@InuSasha binary add-ons are a bit messy at the moment after changes from allwinus. The core changes are merged (which breaks things) and now the next phase of work (fixing the stuff that broke) will be done. It's probably not worth trying to build add-ons for a couple of weeks while things are sorted out :)

File: packages/addons/service/docker/package.mk [hunk 1]
```diff
@@ -29,6 +29,7 @@ PKG_SECTION="service/system"
 PKG_SHORTDESC="Docker is an open-source engine that automates the deployment of any application as a lightweight, portable, self-sufficient container that will run virtually anywhere."
 PKG_LONGDESC="Docker containers can encapsulate any payload, and will run consistently on and between virtually any server. The same container that a developer builds and tests on a laptop will run at scale, in production*, on VMs, bare-metal servers, OpenStack clusters, public instances, or combinations of the above."
 PKG_AUTORECONF="no"
+PKG_SOURCE_DIR="moby-$PKG_VERSION"
```
Comment (escalade) [inline packages/addons/service/docker/package.mk]: I guess this commit wasn't meant for this PR?

Comment (InuSasha) [inline packages/addons/service/docker/package.mk]: @escalade:  you are right. i found it, while i upgrade to gcc7. actual, i had no time to split it into single branches.

**Comment by InuSasha** [comment]
* update List of failing packages/addons
* upload new build artifacts

**Comment by InuSasha** [comment]
only rebase against master

**Comment by InuSasha** [comment]
actual state: all archs builds/fails like master (incl.addons)
but no runtime tests, yet

**Comment by lrusak** [comment]
@InuSasha care to rebase and update to the latest gcc? I need this to be able to update to glibc 2.26. If you don't have time just let me know 👍

**Comment by InuSasha** [comment]
@lrusak i can rebase this. but no runtime tests was made, jet.
Because the open problem is chromium... it does not build with gcc7 and the version upgrade su... is stalled (#1730).

**Comment by zalaare** [comment]
@lrusak If you don't want to wait, I was able to build gcc against libc 2.26 this morning (different project) using the upstreamed patch: https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=b685411208e0aaa79190d54faf945763514706b8

**Comment by InuSasha** [comment]
@lrusak i make a comparison build against master. if the had equal stability i update this pr. if not i will report ;)
i also will drop chromium patches and will pr a wip after merge.

**Comment by InuSasha** [comment]
rebased.
all images are building (except imx6 like master ;) )
addons mostly build (no difference to master)
chromium changes removed

**Comment by InuSasha** [comment]
* Includes #1855, now
* updated binutils, too.
* switched to gcc7.2 RC-1 (waiting for release)

**Comment by InuSasha** [comment]
* update to gcc 7.2 rc.2
* drop kodi gcc7 patch
* rebase to master

**Comment by InuSasha** [comment]
7.2 final was released.
after work, i will update pr.
additional over night & tomorrow, a full build of all archs (except imx6) and all addons.

**Comment by CvH** [comment]
@InuSasha about that `vdr-plugin-live: gcc7 fixes` commit
maybe a better fix https://github.com/jasmin-j/vdr-plugin-live/commit/b86560030911208be738163ecd0e48790c9317a1

**Comment by InuSasha** [comment]
- rebase to origin/master
- add glib #1874 
- replace vdr-plugin-live with better patch
- gcc 7.2 final (switch from tar.bz2 to tar.xz)

**Comment by InuSasha** [comment]
@MilhouseVH added your fix. additional add --with-python=python to target. there are long shebangs, too.

**Comment by CvH** [comment]
@InuSasha you could drop those vdr-satip patch and instead just update the plugin version to `ee88aad` (fix got added upstream at the 2.2.x branch we use)

**Comment by InuSasha** [comment]
suggestion from @CvH added

should ready to merge, now
only chromium is not  building, yet.

**Pull request merged**